libs/environment/src/hooks/use-environment.spec.ts

```diff
@@ -0,0 +1,259 @@
+import { renderHook, waitFor } from '@testing-library/react';
+import { act } from 'react-dom/test-utils';
+import { Networks } from '../types';
+import { useEnvironment } from './use-environment';
+
+const noop = () => {
+  /* no op*/
+};
+
+jest.mock('@vegaprotocol/apollo-client', () => ({
+  createClient: () => ({
+    query: () =>
+      Promise.resolve({
+        data: {
+          statistics: {
+            chainId: 'chain-id',
+            blockHeight: '100',
+            vegaTime: new Date().toISOString(),
+          },
+        },
+      }),
+    subscribe: () => ({
+      // eslint-disable-next-line
+      subscribe: (obj: any) => {
+        obj.next();
+      },
+    }),
+  }),
+}));
+jest.mock('zustand');
+
+global.fetch = jest.fn();
+
+// eslint-disable-next-line
+const setupFetch = (result: any) => {
+  return () => {
+    return Promise.resolve({
+      ok: true,
+      json: () => Promise.resolve(result),
+    });
+  };
+};
+
+const mockEnvVars = {
+  VEGA_ENV: Networks.TESTNET,
+  VEGA_NETWORKS: {
+    DEVNET: 'https://devnet.url',
+    TESTNET: 'https://testnet.url',
+    STAGNET3: 'https://stagnet3.url',
+    MAINNET: 'https://mainnet.url',
+  },
+  VEGA_WALLET_URL: 'https://localhost:1234',
+  ETHEREUM_PROVIDER_URL: 'https://ether.provider',
+  ETHERSCAN_URL: 'https://etherscan.url',
+};
+
+describe('useEnvironment', () => {
+  const env = process.env;
+
+  beforeEach(() => {
+    jest.resetModules();
+    process.env = { ...env };
+
+    process.env['NX_VEGA_ENV'] = mockEnvVars.VEGA_ENV;
```

Comment:
**thought (non-blocking)**: Seeing this made me realise what a big task removing our dependence on env vars is going to be. But staying consistent here is definitely the right thing for now.